src/runtime_src/core/tools/common/tests/TestIPU.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestIPU.h"
+#include "tools/common/XBUtilities.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_hw_context.h"
+#include "xrt/xrt_kernel.h"
+namespace XBU = XBUtilities;
+
+#include <boost/filesystem.hpp>
+
+static constexpr size_t host_app = 1; //opcode
+static constexpr size_t buffer_size = 128;
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+TestIPU::TestIPU()
+  : TestRunner("ipu-verify", 
+                "Run 'Hello World' test on IPU",
+                "1x4.xclbin"){}
+
+boost::property_tree::ptree
+TestIPU::run(std::shared_ptr<xrt_core::device> dev)
+{
+  boost::property_tree::ptree ptree = get_test_header();
```

Comment:
Why is this a built-in test rather than a precompiled executable? I believe we should move towards all tests being executables. It will allow the `validate` command to be much simpler and encourage other parts of AMD to adopt our tool (eventually).

Reply:
I like the idea of writing the tests in the code more now that I thought about it.